### PR TITLE
Add possibility to add Team-based ACL data.

### PR DESCRIPTION
**Pull request by ezhdan-sugarcrm**

### CODE REVIEW

File: install_cli.php [hunk 1]
```diff
@@ -647,6 +664,49 @@ class FakeLogger { public function __call($m, $a) { } }
...
+			'Bugs' => array('delete' => $tbaAccess, 'export' => $tbaAccess),
+			'Cases' => array('delete' => $tbaAccess, 'export' => $tbaAccess),
+			'KBContents' => array('delete' => $tbaAccess, 'export' => $tbaAccess)
+		);
+
+		$ACLRolesData = array();
+		$result = $GLOBALS['db']->query("SELECT id FROM acl_roles WHERE id LIKE 'seed-ACLRoles%'");
+		while($row = $GLOBALS['db']->fetchByAssoc($result)){
+			$ACLRolesData[$row['id']] = $row['id'];
+			$ACLRole = BeanFactory::retrieveBean('ACLRoles', $row['id']);
```
Comment (vadd) [inline install_cli.php]: This retrieve is not necessary and it's very heavy. Take a look at the [setAction](https://github.com/sugarcrm/Mango/blob/master/sugarcrm/modules/ACLRoles/ACLRole.php#L65-L88). It doesn't use `$this`. Thus you could just create an empty bean outside of the `while` and call `setAction` on it:
```php
$roleBean = BeanFactory::getBean('ACLRoles');
while (/* ... */) {
    //...
    $roleBean->setAction($ACLRole->id, $actionId['id'], $access_override);
    //...
}
```

File: install_cli.php [hunk 2]
```diff
@@ -647,6 +664,49 @@ class FakeLogger { public function __call($m, $a) { } }
...
+			foreach ($roleActions as $category => $actions) {
+				foreach ($actions as $name => $access_override) {
+					$queryACL = "SELECT id FROM acl_actions where category='$category' and name='$name'";
```
Comment (vadd) [inline install_cli.php]: It would be much faster to select all IDs outside of the `while` cycle.

File: install_cli.php [hunk 3]
```diff
@@ -647,6 +664,49 @@ class FakeLogger { public function __call($m, $a) { } }
...
+				}
+			}
+		}
+
+		$i = 1;
+		$usersPerRole = is_numeric($_SESSION['tba']) ? $_SESSION['tba'] : 2; //If not specify, set 2 users per role by default.
+		$userRoleId = array_shift($ACLRolesData);
+		$result = $GLOBALS['db']->query("SELECT id from users where id LIKE 'seed-Users%'");
+		while ($row = $GLOBALS['db']->fetchByAssoc($result)) {
+			$GLOBALS['db']->query("INSERT into acl_roles_users(id,user_id,role_id) values('" . create_guid() . "','" . $row['id'] . "','" . $userRoleId . "')");
```
Comment (vadd) [inline install_cli.php]: Would be much faster to group `insert` queries and execute them in chunks.

Comment (ezhdan-sugarcrm) [inline install_cli.php]: @vadd , got your remarks. Thanks!

File: install_cli.php [hunk 4]
```diff
@@ -647,6 +663,49 @@ class FakeLogger { public function __call($m, $a) { } }
...
+		while($row = $GLOBALS['db']->fetchByAssoc($result)){
+			$ACLRolesData[$row['id']] = $row['id'];
+			$ACLRole = BeanFactory::retrieveBean('ACLRoles', $row['id']);
+			foreach ($roleActions as $category => $actions) {
+				foreach ($actions as $name => $access_override) {
+					$queryACL = "SELECT id FROM acl_actions where category='$category' and name='$name'";
+					$resultACL = $GLOBALS['db']->query($queryACL);
+					$actionId = $GLOBALS['db']->fetchByAssoc($resultACL);
+					if (!empty($actionId['id'])) {
+						$ACLRole->setAction($ACLRole->id, $actionId['id'], $access_override);
```
Comment (vadd) [inline install_cli.php]: This operation is also slow: would be much faster to write a separate `setAction` method that does the same, but does it for chunks of data and not one by one. But the pitfall of this approach is that if the behavior of `setAction` will be changed in core we will need to update the logic here.

File: install_cli.php [hunk 5]
```diff
@@ -270,6 +275,9 @@
 	}
 	$modules['Teams'] = $opts['u']*($modules['Teams']/$modules['Users']);
 	$modules['Users'] = $opts['u'];
+	if (isset($opts['tba'])) {
+		$modules['ACLRoles'] = $opts['u'] / 2;
```
Comment (vadd) [inline install_cli.php]: This proportion doesn't seem realistic to me. I would better divide by 10, or even 20.

**Pull request merged**